File: F/1817-finding-the-users-active-minutes/main.go
package main

import "fmt"

func findingUsersActiveMinutes(logs [][]int, k int) []int {
	m := map[int]map[int]int{}
	for i := 0; i < len(logs); i++ {
		_, ok := m[logs[i][0]]
		if !ok {
			m[logs[i][0]] = map[int]int{}
		}
		m[logs[i][0]][logs[i][1]]++
	}
	ret := make([]int, k)
	for _, v := range m {
		ret[len(v)-1]++
	}
	return ret
}

func main() {
	fmt.Println(findingUsersActiveMinutes([][]int{{1, 2}}, 3))
}

File: F/1817-finding-the-users-active-minutes/main.rs
use std::collections::HashMap;
impl Solution {
  pub fn finding_users_active_minutes(logs: Vec<Vec<i32>>, k: i32) -> Vec<i32> {
    let mut m: HashMap<i32, HashMap<i32, i32>> = HashMap::new();

    logs.iter().for_each(|log| {
      m.entry(log[0])
        .or_insert(HashMap::new())
        .entry(log[1])
        .and_modify(|x| (*x) += 1)
        .or_insert(1);
    });

    let mut ret: Vec<i32> = vec![0; k as usize];
    m.iter().for_each(|(_, v)| {
      ret[v.len() - 1] += 1;
    });
    ret
  }
}
